src/EditorFeatures/Core/Implementation/ExtractMethod/AbstractExtractMethodCommandHandler.cs

```diff
@@ -133,12 +133,12 @@ public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContex
                         // wait context. That means the command system won't attempt to show its own wait dialog 
                         // and also will take it into consideration when measuring command handling duration.
                         waitContext.TakeOwnership();
-                        if (!notificationService.ConfirmMessageBox(
+                        if (notificationService.ConfirmMessageBox(
                                 EditorFeaturesResources.Extract_method_encountered_the_following_issues + Environment.NewLine + Environment.NewLine +
                                 string.Join(Environment.NewLine, result.Reasons) + Environment.NewLine + Environment.NewLine +
                                 EditorFeaturesResources.We_can_fix_the_error_by_not_making_struct_out_ref_parameter_s_Do_you_want_to_proceed,
                                 title: EditorFeaturesResources.Extract_Method,
-                                severity: NotificationSeverity.Error))
+                                severity: NotificationSeverity.Error) != true)
```

Comment:
Wait, so we're now showing yes/no/cancel on this, and no and cancel do the same thing?

Reply:
Yes. 'No' already did nothing, which is what we'd want cancel to do. This is not an uncommon UI pattern in general.

Reply:
Yes/No/Cancel is common yes, but having Cancel and No do the same thing is surprising to me! Especially when the cases we already had No meant to cancel the operation!

Reply:
Cancel means "I don't want to think about whether No does what I want". In some cases they do the same thing, but Cancel does a *specific* thing (take no further action).